fhttp/http_server.go

```diff
@@ -74,28 +103,35 @@ func EchoHandler(w http.ResponseWriter, r *http.Request) {
 			r = &nr
 		}
 	}
-	data, err := io.ReadAll(r.Body) // must be done before calling FormValue
-	if err != nil {
-		log.Errf("Error reading %v", err)
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
-	}
-	log.Debugf("Read %d", len(data))
-	handleCommonArgs(w, r)
-	statusStr := r.FormValue("status")
+	reqNum := handleCommonArgs(w, r)
+	statusStr := QueryArg(r, "status")
 	var status int
 	if statusStr != "" {
 		status = generateStatus(statusStr)
 	} else {
 		status = http.StatusOK
 	}
-	gzip := strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") && generateGzip(r.FormValue("gzip"))
+	gzip := strings.Contains(r.Header.Get("Accept-Encoding"), "gzip") && generateGzip(QueryArg(r, "gzip"))
 	if gzip {
 		gwz := NewGzipHTTPResponseWriter(w)
 		defer gwz.Close()
 		w = gwz
 	}
-	size := generateSize(r.FormValue("size"))
+	size := generateSize(QueryArg(r, "size")) // -1 means no size/payload mode
+	var data []byte
+	var err error
+	// Also read the whole input if we're supposed to write something unrelated like size=100
+	h2Mode := (r.ProtoMajor == 2) && (!gzip) && (size == -1)
+	if !h2Mode {
+		h2Mode = false
```

Comment:
This is not necessary.

Reply:
indeed! a leftover! (surprised linters don't say so)

note while this is already functional I haven't finished this MR yet because it doesn't have unit tests. I went into a rabbit hole trying to have fortio http client support the streaming reply in curl subcommand and not submitted that part yet (but got it working in fortio/h2cli first)

Reply:
fixed